src/Bicep.LangServer/Handlers/BicepDeploymentParametersHandler.cs

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.LanguageServer.CompilationManager;
+using Bicep.LanguageServer.Deploy;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    public record BicepDeploymentParametersResponse(List<BicepDeploymentParameter> deploymentParameters, string parametersFileName, string? errorMessage);
+
+    public record BicepDeploymentParameter(string name, string? value, bool isMissingParam, bool isExpression, bool isSecure, ParameterType? parameterType);
+
+    /// <summary>
+    /// Handles getDeploymentParameters LSP request.
+    /// The BicepDeploymentParametersHandler returns information about deployment parameters, parameters file name and error message, if any.
+    /// List of <see cref="BicepDeploymentParameter"/>, included in the response has informtion about the parameter e.g. name , value, if the parameter has
+    /// @secure() decorator, if it's missing default value/is not present in parameters file, is an expression etc
+    /// The above information will be used to display appropriate controls in UI.
+    /// </summary>
+    public class BicepDeploymentParametersHandler : ExecuteTypedResponseCommandHandlerBase<string, string, string, BicepDeploymentParametersResponse>
+    {
+        private readonly ICompilationManager compilationManager;
+
+        public BicepDeploymentParametersHandler(
+            ICompilationManager compilationManager,
+            ISerializer serializer)
+            : base(LangServerConstants.GetDeploymentParametersCommand, serializer)
+        {
+            this.compilationManager = compilationManager;
+        }
+
+        public override Task<BicepDeploymentParametersResponse> Handle(string documentPath, string parametersFilePath, string template, CancellationToken cancellationToken)
+        {
+            var updatedParams = GetUpdatedParams(documentPath, parametersFilePath, template);
+            return Task.FromResult(updatedParams);
+        }
+
+        private BicepDeploymentParametersResponse GetUpdatedParams(string documentPath, string parametersFilePath, string template)
+        {
+            var parametersFileName = GetParameterFileName(documentPath, parametersFilePath);
+
+            try
+            {
+                var parametersFromProvidedParametersFile = GetParametersInfoFromProvidedFile(parametersFilePath);
```

Comment:
nit (something like this):
```suggestion
                var parametersFromProvidedParametersFile = GetParametersInfoFromProvidedFile(parametersFilePath) ?? Dictionary<string, dynamic>.Empty();
```

Reply:
Linq version doesn't work for dictionary. I updated here to return an empty dictionary:
https://github.com/Azure/bicep/commit/a28f680d0f2256f05e8c02f2923fe3ce2f7df05c